Source/items.cpp

```diff
@@ -961,7 +961,8 @@ BOOL ItemSpaceOk(int i, int j)
 {
 	int oi;
 
-	if (i < 0 || i >= MAXDUNX || j < 0 || j >= MAXDUNY)
+	// BUGFIX: Check `i + 1 >= MAXDUNX` and `j + 1 >= MAXDUNY` (applied)
+	if (i < 0 || i + 1 >= MAXDUNX || j < 0 || j + 1 >= MAXDUNY)

```

Comment:
This would probably be better since it definitely can be baked by the compiler to avoid doing math at runtime.
```suggestion
	if (i < 0 || i >= MAXDUNX - 1 || j < 0 || j >= MAXDUNY - 1)
```

Reply:
No difference even with the default compiler options:
https://godbolt.org/z/-ZUjx7
https://godbolt.org/z/c3NPgZ